crates/ruff_linter/src/rules/ruff/rules/unnecessary_round_cast.rs

```diff
@@ -0,0 +1,120 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Expr, ExprCall, ExprNumberLiteral, Number};
+use ruff_python_semantic::SemanticModel;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for `int` conversions of `round()` calls
+/// with the second argument being either omitted, `0` or `None`.
+///
+/// ## Why is this bad?
+/// Such a `round()` call already returns an integer,
+/// so calling `int()` is unnecessary.
+/// Additionally, the second argument can be omitted if it is `0` or `None`.
+///
+/// ## Known problems
+/// This rule is prone to false positives due to type inference limitations.
+///
+/// ## Example
+///
+/// ```python
+/// int(round(foo, 0))
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// round(foo)
+/// ```
+#[derive(ViolationMetadata)]
+pub(crate) struct UnnecessaryRoundCast;
+
+impl AlwaysFixableViolation for UnnecessaryRoundCast {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "The result of `round()` is already an integer".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        "Replace with `round(...)`".to_string()
+    }
+}
+
+/// RUF046
+pub(crate) fn unnecessary_round_cast(checker: &mut Checker, call: &ExprCall) {
+    let semantic = checker.semantic();
+
+    let Some(argument) = single_argument_to_int_call(semantic, call) else {
+        return;
+    };
+
+    let Some(argument) = first_argument_to_round_to_int_call(semantic, argument) else {
+        return;
+    };
+
+    let new = format!("round({})", checker.locator().slice(argument));
+    let edit = Edit::range_replacement(new, call.range);
+    let fix = Fix::safe_edit(edit);
+
+    let diagnostic = Diagnostic::new(UnnecessaryRoundCast, call.range);
+
+    checker.diagnostics.push(diagnostic.with_fix(fix));
+}
+
+fn single_argument_to_int_call<'a>(
+    semantic: &SemanticModel,
+    call: &'a ExprCall,
+) -> Option<&'a Expr> {
+    let ExprCall {
+        func, arguments, ..
+    } = call;
+
+    if !semantic.match_builtin_expr(func, "int") {
+        return None;
+    }
+
+    if !arguments.keywords.is_empty() {
+        return None;
+    }
+
+    let [argument] = &*arguments.args else {
+        return None;
+    };
+
+    Some(argument)
+}
+
+fn first_argument_to_round_to_int_call<'a>(
+    semantic: &SemanticModel,
+    expr: &'a Expr,
+) -> Option<&'a Expr> {
+    let Expr::Call(ExprCall {
+        func, arguments, ..
+    }) = expr
+    else {
+        return None;
+    };
+
+    if !semantic.match_builtin_expr(func, "round") {
+        return None;
+    }
+
+    if arguments.len() != 1 && arguments.len() != 2 {
+        return None;
+    }
```

Comment:
Nit: The 0 case is handled below by `find_argument("number", 0)?` and searching an empty `arguments` list is cheap

```suggestion
    if arguments.len() >2 {
        return None;
    }
```